### ISSUE TITLE
Feature request: type field

### ISSUE THREAD

**Issue by adamsilkcm** [murphypetercl/ansible-mdgen#29]
Currently, the @var lines support only a name and a description.  I would like to also use this line to optionally define the variable type, e.g. string, list, dict, without it just being a part of the description.

Please let me know if you need further information.

**Comment by murphypetercl** [comment]
Hi Adam, how about something like this:
I could define a configuration parameter called output_variable_types which is set to False by default. You would set this to True in your config file to enable the functionality. Then add the type as follows in your defaults/vars files with the @type annotation within brackets:
\# @var: minio_server_datadirs: (@type: String) Minio server data directory
minio_server_datadirs: /var/lib/minio

Which would then be displayed like this. Basically placing this along with the Where Referenced information in a 2 row table format.


![image](https://user-images.githubusercontent.com/73845971/112029920-a1414600-8b31-11eb-8690-d3858eb7498c.png)

Let me know what you think.

**Comment by adamsilkcm** [comment]
My thoughts on future-proofing this would be to have a list of fields in the config file that would be the columns for the output (or dynamically generate it based on fields defined in all the @var lines for a given role).  Not sure how unwieldy that would get parsing (or writing) a single line for n fields, though. Maybe parsing yaml that's commented out and has markers for the start and end of what ansible-mdgen should parse for it, like:
```
# @var: 
# minio_server_datadirs:
#   type: string
#   description: Minio server data directory
#   vault_required: true
# @var_end
```
If that's overkill, no worries, just thinking that if there's a case for making a column for type, there's probably a case for all sorts of other things.

**Comment by murphypetercl** [comment]
Yes that makes sense. Started work on it over the weekend.
For backwards compatibility, will keep the existing structure of "# @var: <var_name>: <var_description>" but allow user to alternatively enter details in the format you suggested. 
Any fields in the @var will be added as columns (except for description - will keep that displayed in the main section). Will also add a transpose configuration option to transpose the table (for cases where there are many fields and displaying in one row may not be the best visually).

**Comment by murphypetercl** [comment]
Hi Adam, those changes are now available in the latest version 0.1.0.

**Comment by adamsilkcm** [comment]
Great and thank you so much!
